.github/workflows/generate_new_client.yaml

```diff
@@ -0,0 +1,143 @@
+        id: generation
+        run: |
+          set -x
+          arguments="--api_shortname=\"${API_SHORTNAME}\" \
+          --proto-path=\"${PROTO_PATH}\" \
+          --name-pretty=\"${NAME_PRETTY}\" \
+          --product-docs=\"${PRODUCT_DOCS}\" \
+          --api-description=\"${API_DESCRIPTION}\""
+
+          # helper function that appends a python argument only if specified in the GH action inputs
+          append_argument() {
+            py_arg=$1
+            # env vars look exactly like new-client arguments but uppercase + underscores
+            env_name=$(echo "${py_arg}" | sed 's/-/_/g' | sed -e 's/\([a-z]\)/\U\1/g')
+            if [[ -n "${!env_name}" ]]; then
+              # $(echo) is redundant but it works around a syntax highlighting problem in vim
+              arguments=$(echo "${arguments} --${py_arg}=\"${!env_name}\"")
+            fi
+          }
+
+          declare -a optional_args=('transport' 'destination-name' 'distribution-name' 'group-id' 'rest-docs' 'rpc-docs')
+
+          for python_argument in "${optional_args[@]}"; do
+            append_argument "${python_argument}"
+          done
+          echo "::set-output name=new_library_args::${arguments}"
+          echo "${arguments} --googleapis-gen-url=\"${GOOGLEAPIS_GEN_URL}\"" | xargs python generation/new_client/new-client.py generate
+        env:
+          GOOGLEAPIS_GEN_URL: https://yoshi-approver:${{ secrets.YOSHI_CODE_BOT_TOKEN }}@github.com/googleapis/googleapis-gen.git
+          API_SHORTNAME: ${{ github.event.inputs.api_shortname }}
+          NAME_PRETTY: ${{ github.event.inputs.name_pretty }}
+          PROTO_PATH: ${{ github.event.inputs.proto_path }}
+          PRODUCT_DOCS: ${{ github.event.inputs.product_docs }}
+          REST_DOCS: ${{ github.event.inputs.rest_docs }}
+          RPC_DOCS: ${{ github.event.inputs.rpc_docs }}
+          API_DESCRIPTION: ${{ github.event.inputs.api_description }}
+          TRANSPORT: ${{ github.event.inputs.transport }}
+          DESTINATION_NAME: ${{ github.event.inputs.destination_name }}
+          DISTRIBUTION_NAME: ${{ github.event.inputs.distribution_name }}
+      - name: Push to branch and create PR
+        run: |
+          set -x
+          [ -z "`git config user.email`" ] && git config --global user.email "${USERNAME:-script}@google.com"
+          [ -z "`git config user.name`" ] && git config --global user.name "${USERNAME:-script}"
+
+          # create and push to branch in origin
+          # random_id allows multiple runs of this workflow
+          random_id=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 5; echo)
+          branch_name="new-library/${{ github.event.inputs.api_shortname }}-${random_id}"
+          git checkout -b "${branch_name}"
+          git add --all
+          git commit -m "feat: [${API_SHORTNAME}] new module for ${API_SHORTNAME}
+
+
+          \`\`\`
+          python generation/new_client/new-client.py generate ${GENERATION_ARGUMENTS}
+          \`\`\`"

```

Comment:
Can you also add a line that says this pull request was created via this workflow?

https://github.com/googleapis/google-cloud-java/actions/workflows/generate_new_client.yaml